src/VSCodeUI.ts

```diff
@@ -53,6 +54,47 @@ export namespace VSCodeUI {
         }
     }
 
+    export function setupEnvironment(terminal: Terminal): void {
+        // do this first so it can be overridden if desired
+        setJavaHomeIfAvailable(terminal);
+
+        type EnvironmentSetting = {
+            environmentVariable: string;
+            value: string;
+        };
+
+        const environmentSettings: EnvironmentSetting[] = workspace.getConfiguration("maven").get("terminal.customEnv");
+        environmentSettings.forEach((s: EnvironmentSetting) => {
+            terminal.sendText(getEnvironmentVariableCommand(s.environmentVariable, s.value), true);
+        });
+    }
+
+    export function setJavaHomeIfAvailable(terminal: Terminal): void {
+        const javaHome: string = workspace.getConfiguration("java").get<string>("home");
+        const useJavaHome: boolean = workspace.getConfiguration("maven").get<boolean>("terminal.use.java.home");
+        if (useJavaHome && javaHome) {
+            terminal.sendText(getEnvironmentVariableCommand("JAVA_HOME", javaHome), true);
+        }
+    }
+
+    export function getEnvironmentVariableCommand(variable: string, value: string): string {
```

Comment:
confusing name, might be `setEnvironmentVariableCommand`? Is it a typo?

Reply:
It is a "get" that returns a "set" command string (ha ha).  I'll figure out a different name.